frontend/components/puzzle4.jsx

```diff
@@ -34,9 +35,23 @@ class Puzzle4 extends React.Component {
     });
   }
 
+  submitAnswer() {
+    let answer = this.state.value;
+    let gameState = this.props.gameState;
+
+    // base case
+    if (Object.keys(gameState).length === 0) {
+      // TODO: can you explain why "if (gameState === {})" doesn't work?
+      console.log("no other puzzles have been solved");
+      return;
+    }
+
+    if (gameState[1]["solved"] && gameState[2]["solved"] && gameState[3]["solved"]) {
```

Comment:
might be easier to use the method here: https://github.com/iamsammak/thehuntgame/pull/55/files#diff-4a226b065b15610552fd2c9bbdb37aa3

Reply:
not sure how that getCurrentStage helper method is helpful here...I probably don't follow what that method is supposed to do

Reply:
that method should be available now. you can do
```
stage = getCurrentStage(gameState);
// Stage 4 implies 1,2,3 are solved
if (stage === 4) {
  // do stuff
}
```